# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/JitScript.h
# Commit: dab916754456
# Full Hash: dab916754456373246c85d658fde0520c7246865
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-07-23 03:20:25
# Regressor Bug: 1646378
# File Overlap Count: 1
# Description:
#   Bug 1646378: Invalidate the root script in MaybeInvalidateWarp r=jandem
#   
#   When we replace a stub in an inlined ICScript, we want to invalidate the script into which it was inlined, rather than the original copy of the inlined script.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84431
# ==============================================================================

diff -r 86db033f2b35 -r dab916754456 js/src/jit/JitScript.h
--- a/js/src/jit/JitScript.h	Wed Jul 22 11:31:25 2020 +0000
+++ b/js/src/jit/JitScript.h	Wed Jul 22 21:57:45 2020 +0000
@@ -99,7 +99,9 @@
         warmUpCount_(warmUpCount),
         endOffset_(endOffset) {}
 
-  JitScript* jitScript() { return jitScript_; }
+  JitScript* jitScript() const { return jitScript_; }
+
+  bool isInlined() const;
 
   MOZ_MUST_USE bool initICEntries(JSContext* cx, JSScript* script);
 
@@ -723,7 +725,7 @@
 
   bool hasInliningRoot() const { return !!inliningRoot_; }
   InliningRoot* inliningRoot() const { return inliningRoot_.get(); }
-  InliningRoot* getOrCreateInliningRoot(JSContext* cx);
+  InliningRoot* getOrCreateInliningRoot(JSContext* cx, JSScript* script);
   void clearInliningRoot() { inliningRoot_.reset(); }
 };
 